rPackages.pkgdepends: fix compile error

Kupac opened the pull request:
## Description of changes

Needed a bit of shebangs patching

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Commit:
rPackages.pkgdepends: fix compile error

Locate patchShebangs correctly

Co-authored-by: Justin Bedő <cu@cua0.org>

b-rodrigues commented:
Tested on linux-aarch64, looking good!

```
> pkgdepends::current_config()
# pkg config
## build_vignettes
<default>
[1] FALSE

## sysreqs_verbose
<default>
[1] FALSE

## sysreqs_db_update
<default>
[1] TRUE

## metadata_cache_dir
<default>
[1] "/run/user/1000/RtmpUU5QPy/file5e2db1a24ef"

## platforms
<default>
[1] "aarch64-unknown-linux-gnu-ubuntu-23.10"
[2] "source"                                

## goal
<default>
[1] "unknown"

## r_versions
<default>
[1] "4.3.2"

## cache_dir
<default>
[1] "/run/user/1000/RtmpUU5QPy/file5e2d32bb2ffa"

## library
<default>
NULL

## metadata_update_after
<default>
Time difference of 24 hours

## include_linkingto
<default>
[1] FALSE

## sysreqs_rspm_repo_id
<default>
[1] "1"

## sysreqs_update
<default>
[1] FALSE

## package_cache_dir
<default>
NULL

## sysreqs_rspm_url
<default>
[1] "https://packagemanager.posit.co"

## sysreqs_sudo
<default>
[1] TRUE

## sysreqs_db_update_timeout
<default>
Time difference of 5 secs

## sysreqs_lookup_system
<default>
[1] TRUE

## sysreqs_platform
<default>
[1] "aarch64-unknown-linux-gnu-ubuntu-23.10"

## dependencies
<default>
[1] "Depends"   "Imports"   "LinkingTo"

## sysreqs
<default>
[1] FALSE

## sysreqs_dry_run
<default>
[1] FALSE

## windows_archs
<default>
[1] "prefer-x64"

## use_bioconductor
<default>
[1] TRUE

## cran_mirror
<default>
[1] "https://cran.rstudio.com"

```

jbedo commented on pkgs/development/r-modules/default.nix:
```diff
@@ -1436,6 +1436,12 @@ let
     });
+
+    pkgdepends = old.pkgdepends.overrideAttrs (attrs: {
+      preConfigure = ''
```
```suggestion
      postPatch = ''
```

b-rodrigues commented:
Can you squash the commits?

jbedo committed:
rPackages.igraph: fix build

Kupac commented:
@jbedo I addressed your comment, tried to resolve-unresolve-resolve comment, rebase and force push twice, but it still says merging is blocked. Is there a button somewhere I don't see? how can I resolve the issue to allow merging?

jbedo committed:
rPackages.Rsamtools: fix build

jbedo committed:
rPackages: CRAN and BioC update

jbedo committed:
rPackages.Rhdf5lib: use newer hdf5

jbedo committed:
rPackages.Rhdf5lib: downgrade hdf5

Major dependent rPackages.rhdf5 is not compatible with newer version

b-rodrigues committed:
rPackages.cartogramR: fixed build

b-rodrigues committed:
rPackages.image_textlinedetector: fixed build

jbedo committed:
rPackages.VariantAnnotation: fix build

jbedo committed:
rPackages.Rhdf5lib: enable c++ bindings

jbedo committed:
rPackages.bamsignals: fix build

jbedo committed:
rPackages.maftools: fix build

jbedo committed:
rPackages.RcppCGAL: fix build

jbedo committed:
rPackages.hdf5r: fix build

jbedo committed:
Merge pull request #299831 from b-rodrigues/fix_rpackages.textlinedetector

rPackages.image_textlinedetector: fixed build

jbedo committed:
Merge pull request #299827 from b-rodrigues/fix_rpackages.cartogramR

rPackages.cartogramR: fixed build
